src/utils/telemetry.ts

```diff
@@ -1,5 +1,49 @@
 /* eslint-disable no-console, max-classes-per-file */
 import { Logger } from './logger';
+import pkg from '../../package.json';
+
+/**
+ * Constants used for telemetry event names. These string values should never change
+ */
+export const telemetryType = {
+  initializationSpan: 'dlo_init_span',
+  ruleCollectionSpan: 'dlo_rule_collection_span',
+  ruleRegistrationSpan: 'dlo_rule_registration_span',
+  ruleCount: 'dlo_rule_count',
+  handleEventSpan: 'dlo_handle_event_span',
+  clientError: 'dlo_client_error',
+};
+
+/**
+ * Constants used for error types in client error telemetry. These string values should
+ * never change
+ */
+export const errorType = {
```

Comment:
I elected to have separate error constants for telemetry rather than overloading e.g. `LogMessageType.OperatorError` so these could evolve independently. Logging message types could be changed with less friction; telemetry error types should really be pinned down as exporters may implement logic based on those values.

In the current implementation, I'm also explicitly calling `Telemetry.count` after we call `Logger.getInstance().error`. We could consider always calling `Telemetry.count` within `Logger.getInstance().error`, at which point I'd probably lean toward overloading `LogMessageType` or maybe even extracting that to a dedicated `ErrorType` concept which is used by both logging and telemetry. Interested in feedback on this.

Reply:
I think we'll need to revisit how our current logger works at some point - currently we're consuming customer FS.event quota. I can see a future where both logging and metrics are collapsed into a common framework that uses the same telemetry provider - maybe one that implements both the current otel-based interfaces as well as a "logger" interface. Idunno, just riffing; I could be way off-base about implementation.

I think the implementation as-is is fine for this PR.